File: scala/problem1.scala
object Main {
  def main(args: Array[String]) {
    println(sumDivisorOfThreeOrFive(10))
    println(sumDivisorOfThreeOrFive(1000))
  }

  def sumDivisorOfThreeOrFive(n: Int): Int = {
    return (1 to n).filter(n => n % 3 == 0 || n % 5 == 0).sum
  }
}

File: rust/problem1.rs
fn main() {
    println!("{}", sum_divisor_of_three_or_five(10));
    println!("{}", sum_divisor_of_three_or_five(1000));
}

fn sum_divisor_of_three_or_five(n: u32) -> u32 {
    (1..n).filter(|x| x % 3 == 0 || x % 5 == 0).fold(0, |current, x| current + x)
}
